scripts/generate_can.py

```diff
@@ -103,82 +103,83 @@ def camel_to_snake_case(value):
-                AUTOGEN_msg_map[msg_snake_upper].append(signal_snake_upper)
-
-                length = signal['length']
-                is_signed = signal['is_signed']
-
-                # NOTE: it is assumed that min/max bounds do not cover SNA values
-
-                # if given both scale/offset and min/max
-                if 'scale' in signal and 'offset' in signal and 'min' in signal and 'max' in signal:
-                    scale = signal['scale']
-                    offset = signal['offset']
-                    sig_min = signal['min']
-                    sig_max = signal['max']
-
-                # else if only given scale/offset (no min/max)
-                elif 'scale' in signal and 'offset' in signal:
-                    scale = signal['scale']
-                    offset = signal['offset']
-
-                    # calculate min/max based on scale/offset and length
-                    sig_min = - (2**(length-1)-1) * scale + \
-                        offset if is_signed else offset
-                    sig_max = (2**(length-1)-1) * scale + \
-                        offset if is_signed else (2**length-2) * scale + offset
-
-                # else if only given min/max (no scale/offset)
-                elif 'min' in signal and 'max' in signal:
-                    sig_min = signal['min']
-                    sig_max = signal['max']
-
-                    # calculate scale/offset based on min/max and length
-                    scale = (sig_max - sig_min) / (2**length - 2)
-                    offset = sig_min + (2**(length-1)-1) * \
-                        scale if is_signed else sig_min
-
-                else:
-                    print('Missing fields for signal "' + signal_name +
-                        '" in message "' + message_name + '"!')
-                    print('Supply either signal scale/offset or signal min/max')
-                    print('Script terminated.')
-                    exit()
-
-                # add these missing fields to the yaml
-                signal['scale'] = scale
-                signal['offset'] = offset
-                signal['min'] = sig_min
-                signal['max'] = sig_max
-
-                signals.append(
-                    cantools.database.can.Signal(
-                        name=signal_name,
-                        start=startbit,
-                        length=signal['length'],
-                        byte_order='little_endian',
-                        is_signed=signal['is_signed'],
-                        scale=signal['scale'],
-                        offset=signal['offset'],
-                        minimum=signal['min'],
-                        maximum=signal['max'],
-                        unit=signal['unit'],
-                        choices=signal['values'] if 'values' in signal else None,
-                        comment=signal['comment'],
-                        receivers=message['receivers']
+            if 'signals' in message:
```

Comment:
@jessieramseyer plz dont introduce whitespace and formatting changes. makes diff harder to review

@cindyli-13 should add a quick ci stage to normalize formatting for python

Reply:
Yeah I agree, the diff is really hard to look at. How would I fix it? I'm not sure which formatting changes I made

Reply:
its whitespace changes. the number of spaces in the file per indent level has changed